homeassistant/components/cookidoo/todo.py

```diff
@@ -0,0 +1,179 @@
+        """Update an ingredient to the To-do list.
+
+        Cookidoo ingredients can be changed in state, but not in summary or description. This is currently not possible to distinguish in home assistant and just fails silently.
+        """
+        try:
+            assert item.uid
+            await self.coordinator.cookidoo.edit_ingredients_ownership(
+                [
+                    CookidooItem(
+                        {
+                            "id": item.uid,
+                            "name": "<nil>",
+                            "description": "<nil>",
+                            "isOwned": item.status == TodoItemStatus.COMPLETED,
+                        }
+                    )
+                ]
+            )
+        except CookidooException as e:
+            raise HomeAssistantError(
+                translation_domain=DOMAIN,
+                translation_key="todo_update_item_failed",
+                translation_placeholders={"name": item.summary or ""},
+            ) from e
+
+        await self.coordinator.async_refresh()
+
+
+class CookidooAdditionalItemTodoListEntity(CookidooBaseEntity, TodoListEntity):
+    """A To-do List representation of the additional items in the Cookidoo Shopping List."""
+
+    _attr_translation_key = "additional_item_list"
+    _attr_supported_features = (
+        TodoListEntityFeature.CREATE_TODO_ITEM
+        | TodoListEntityFeature.UPDATE_TODO_ITEM
+        | TodoListEntityFeature.DELETE_TODO_ITEM
+    )
+
+    def __init__(self, coordinator: CookidooDataUpdateCoordinator) -> None:
+        """Initialize the entity."""
+        super().__init__(coordinator)
+        self._attr_unique_id = f"{coordinator.config_entry.entry_id}_additional_items"
+
+    @property
+    def todo_items(self) -> list[TodoItem]:
+        """Return the todo items."""
+
+        return [
+            TodoItem(
+                uid=item["id"],
+                summary=item["name"],
+                status=(
+                    TodoItemStatus.COMPLETED
+                    if item["isOwned"]
+                    else TodoItemStatus.NEEDS_ACTION
+                ),
+            )
+            for item in self.coordinator.data[TODO_ADDITIONAL_ITEMS]
+        ]
+
+    async def async_create_todo_item(self, item: TodoItem) -> None:
+        """Add an item to the To-do list."""
+
+        try:
+            assert item.summary
+            await self.coordinator.cookidoo.add_additional_items([item.summary])
+        except CookidooException as e:
+            raise HomeAssistantError(
+                translation_domain=DOMAIN,
+                translation_key="todo_save_item_failed",
+                translation_placeholders={"name": item.summary or ""},
+            ) from e
+
+        await self.coordinator.async_refresh()
+
+    async def async_update_todo_item(self, item: TodoItem) -> None:
+        """Update an item to the To-do list."""
+
+        try:
+            assert item.uid
```

Comment:
same

Reply:
I like this more and have used it in other integrations such as `bring` and `swiss_public_transport`

```
if TYPE_CHECKING:
   assert asdf
```